File: examples/python/messages/upload_attachment.py
# python -m messages.upload_attachment
import mimetypes
from pathlib import Path
import requests
import os
from utils.config import API_KEY, BASE_URL

TEST_IMAGE_PATH = os.path.join(
    os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 
    'inputs', 
    'cat.jpg'
)

def main(file_path: str | Path) -> dict:
    file_path = Path(file_path)
    
    headers = {'Authorization': f'Api-Key {API_KEY}'}
    mime_type, _ = mimetypes.guess_type(file_path)

    with open(file_path, 'rb') as file:
        files = {'file': (file_path.name, file, mime_type)}
        response = requests.post(f'{BASE_URL}attachments/', headers=headers, files=files)

    print(response.json())


if __name__ == '__main__':
    main(TEST_IMAGE_PATH)

File: examples/csharp/messages/upload_attachment.cs
using System;
using System.IO;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text.Json;
using System.Threading.Tasks;
using Utils;

namespace MaiAgentExamples.Messages
{
    public static class UploadAttachment
    {
        public static string TEST_IMAGE_PATH = Path.Combine(
            Directory.GetParent(Directory.GetCurrentDirectory())?.Parent?.Parent?.FullName ?? "",
            "inputs",
            "cat.jpg"
        );

        public static async Task Main(string[] args)
        {
            var filePath = args.Length > 0 ? args[0] : TEST_IMAGE_PATH;
            await UploadFile(filePath);
        }

        private static async Task UploadFile(string filePath)
        {
            if (!File.Exists(filePath))
            {
                Console.WriteLine($"File not found: {filePath}");
                return;
            }

            using var httpClient = new HttpClient();
            httpClient.DefaultRequestHeaders.Authorization =
                new AuthenticationHeaderValue("Api-Key", Config.API_KEY);

            using var form = new MultipartFormDataContent();
            var fileStream = File.OpenRead(filePath);
            var fileContent = new StreamContent(fileStream);

            // Set content type based on file extension
            var extension = Path.GetExtension(filePath).ToLower();
            fileContent.Headers.ContentType = extension switch
            {
                ".jpg" or ".jpeg" => new MediaTypeHeaderValue("image/jpeg"),
                ".png" => new MediaTypeHeaderValue("image/png"),
                ".gif" => new MediaTypeHeaderValue("image/gif"),
                ".pdf" => new MediaTypeHeaderValue("application/pdf"),
                _ => new MediaTypeHeaderValue("application/octet-stream")
            };

            form.Add(fileContent, "file", Path.GetFileName(filePath));

            var response = await httpClient.PostAsync($"{Config.BASE_URL}attachments/", form);
            response.EnsureSuccessStatusCode();

            var responseContent = await response.Content.ReadAsStringAsync();
            var jsonResponse = JsonSerializer.Deserialize<JsonElement>(responseContent);

            Console.WriteLine(JsonSerializer.Serialize(jsonResponse, new JsonSerializerOptions { WriteIndented = true }));
        }
    }
}
